torchvision/transforms/functional_pil.py

```diff
@@ -261,8 +261,7 @@ def pad(img, padding, fill=0, padding_mode="constant"):
         raise ValueError("Padding mode should be either constant, edge, reflect or symmetric")
 
     if padding_mode == "constant":
-        if isinstance(fill, numbers.Number):
```

Comment:
@vfdev-5 Is it correct that we only perform this for `padding_mode == "constant"`?

Reply:
I get it know: all other `padding_mode`s use values from the image itself and thus do not need a `fill` :facepalm: